### ISSUE TITLE
Jobs killed by gnt-job cancel --kill can still hold the BGL

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#1027]
Originally reported of Google Code with ID 971.

```
The new `gnt-job cancel --kill` feature doesn't yet tell WConfD that jobs are dead, leading to jobs not being released immediately (only after the 5 minutes timeout).

To reproduce:

    gnt-debug delay --submit 100
    gnt-job cancel --kill --yes-do-it JOBID
    gnt-debug locks

Where JOBID will be the owner of `cluster/BGL`.
```
Originally added on 2014-10-08 11:58:17 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-10-08 15:33:59 +0000 UTC.

Changed State: Started

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 5025d56c111a08e0feb990f7b18d08b57701bdc9
Author: Klaus Aehlig <aehlig@google.com>
Date:   Wed Oct 8 18:50:46 2014 +0200

    After killing a job inform WConfD
    
    ...that now is a good time to check if some resource owners
    have died.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Niklas Hambuechen <niklash@google.com>

commit a03c883a2a5e9ec53f7771383109a98da0ccad15
Author: Klaus Aehlig <aehlig@google.com>
Date:   Wed Oct 8 18:39:19 2014 +0200

    Document gnt-debug wconfd cleanuplocks in the man page
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Niklas Hambuechen <niklash@google.com>

commit d0024c5bd5f09199c9d11f652d375dbf6fd6506b
Author: Klaus Aehlig <aehlig@google.com>
Date:   Wed Oct 8 18:30:29 2014 +0200

    Add gnt-debug wconfd cleanuplocks
    
    ...which asks WConfD to clean up all stale locks.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Niklas Hambuechen <niklash@google.com>

commit c48afe5319f61e6a7c416ed2f7fc06f101b1a962
Author: Klaus Aehlig <aehlig@google.com>
Date:   Wed Oct 8 18:08:34 2014 +0200

    Export cleanupLocks through WConfD
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Niklas Hambuechen <niklash@google.com>

commit 90e3e9e6f5439a35f1af74f1df7e9dd6f40112e1
Author: Klaus Aehlig <aehlig@google.com>
Date:   Wed Oct 8 18:04:13 2014 +0200

    Extract cleanupLocks from cleanupLocksTask
    
    Extract a function to clean up locks once. This will allow
    for out-of-schedule death detection.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Niklas Hambuechen <niklash@google.com>

commit 636807f4655ae1feaa5e5a60fb7764a30583ec8d
Author: Klaus Aehlig <aehlig@google.com>
Date:   Wed Oct 8 18:18:24 2014 +0200

    Directly call cleanup function from Persistent
    
    ...so that that module becomes independent of Core.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Niklas Hambuechen <niklash@google.com>

```
Originally added on 2014-10-08 17:30:39 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:37:03 +0000 UTC.

Changed State: Released
